transformer/kubernetes/argocdtransformer.go

```diff
@@ -0,0 +1,155 @@
+	return nil, nil
+}
+
+// Transform transforms artifacts understood by the transformer
+func (t *ArgoCD) Transform(newArtifacts []transformertypes.Artifact, alreadySeenArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	logrus.Tracef("ArgoCD transformer Transform start")
+	defer logrus.Tracef("ArgoCD transformer Transform end")
+
+	pathMappings := []transformertypes.PathMapping{}
+	createdArtifacts := []transformertypes.Artifact{}
+
+	for _, newArtifact := range newArtifacts {
+		if newArtifact.Type != irtypes.IRArtifactType {
+			continue
+		}
+		var ir irtypes.IR
+		if err := newArtifact.GetConfig(irtypes.IRConfigType, &ir); err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", ir, err)
+			continue
+		}
+		var clusterConfig collecttypes.ClusterMetadata
+		if err := newArtifact.GetConfig(ClusterMetadata, &clusterConfig); err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", clusterConfig, err)
+			continue
+		}
+		ir.Name = newArtifact.Name
+		preprocessedIR, err := irpreprocessor.Preprocess(ir)
+		if err != nil {
+			logrus.Errorf("Unable to prepreocess IR : %s", err)
+		} else {
+			ir = preprocessedIR
+		}
+		resources := []apiresource.IAPIResource{
+			new(apiresource.ArgoCDApplication),
+		}
+		deployCICDDir := t.ArgoCDConfig.OutputPath
+		tempDest := filepath.Join(t.Env.TempPath, deployCICDDir)
+		logrus.Debugf("Generating ArgoCD yamls for CI/CD")
+		enhancedIR := t.setupEnhancedIR(ir, t.Env.GetProjectName())
+		files, err := apiresource.TransformIRAndPersist(enhancedIR, tempDest, resources, clusterConfig)
+		if err != nil {
+			logrus.Errorf("failed to transform and persist IR. Error: %q", err)
+			continue
+		}
+		for _, file := range files {
+			destPath, err := filepath.Rel(t.Env.TempPath, file)
+			if err != nil {
+				logrus.Errorf("failed to make the yaml path %s relative to the temporary directory %s . Error: %q", file, t.Env.TempPath, err)
+				continue
+			}
+			pathMappings = append(pathMappings, transformertypes.PathMapping{
+				Type:     transformertypes.DefaultPathMappingType,
+				SrcPath:  file,
+				DestPath: destPath,
+			})
+		}
+		createdArtifact := transformertypes.Artifact{
+			Name: t.Config.Name,
+			Type: artifacts.KubernetesYamlsArtifactType,
+			Paths: map[transformertypes.PathType][]string{
+				artifacts.KubernetesYamlsPathType: {deployCICDDir},
+			},
+		}
+		createdArtifacts = append(createdArtifacts, createdArtifact)
+		logrus.Debugf("ArgoCD generated %d new objects", len(files))
+	}
+
+	return pathMappings, createdArtifacts, nil
+}
+
+// setupEnhancedIR returns EnhancedIR containing ArgoCD components
+func (t *ArgoCD) setupEnhancedIR(oldir irtypes.IR, projectName string) irtypes.EnhancedIR {
+	ir := irtypes.NewEnhancedIRFromIR(oldir)
+	// Prefix the project name and make the name a valid k8s name.
+	p := func(baseName string) string {
```

Comment:
Why do we need a function for this?

Reply:
I made it similar to Tekton. ArgoCD has other resources like `Project` which we may need to generate in the future.